Encoder/JWTEncoderInterface.php

```diff
@@ -14,21 +14,23 @@
 {
     /**
      * @param array $data
+     * @param array $header
      *
      * @return string the encoded token string
      *
      * @throws JWTEncodeFailureException If an error occurred while trying to create
      *                                   the token (invalid crypto key, invalid payload...)
      */
-    public function encode(array $data);
+    public function encode(array $data, array $header = []);
```

Comment:
Adding an argument here still breaks BC, even if optional. The signature in concrete implementations must match the interface one. What we need is to introduce a `HeaderAwareEncoderInterface`, make our built-in encoders implement it and check that the passed encoder does implement it before relying on the new argument in JWTManager.